www/src/components/sidebar/body.js

```diff
@@ -0,0 +1,104 @@
+import React from "react"
+
+import Section from "./section"
+import presets, { colors } from "../../utils/presets"
+import { options, rhythm, scale } from "../../utils/typography"
+
+class SidebarBody extends React.Component {
+  constructor(props, context) {
+    super(props, context)
+
+    let state = {
+      openSectionHash: {},
+    }
+
+    if (props.sectionList.length === 1) {
+      state.openSectionHash[props.sectionList.title] = true
+    } else {
+      props.sectionList.forEach(child => {
+        state.openSectionHash[child.title] =
+          props.defaultActiveSection.title === child.title ? true : false
+      })
+    }
+
+    this.state = state
+  }
+
+  _toggleSection(section) {
+    const openSectionHash = this.state.openSectionHash
+    this.setState({
+      openSectionHash: {
+        ...openSectionHash,
+        [section.title]: !openSectionHash[section.title],
+      },
+    })
+  }
+
+  _expandAll = () => {
+    const { openSectionHash } = this.state
+    Object.keys(openSectionHash).forEach(k => (openSectionHash[k] = true))
```

Comment:
this mutates state directly -- we need to make a copy of the object and then call `setState`

Reply:
eg `const newOpenSectionHash = { ...openSectionHash }`